### ISSUE TITLE
[V2][Android] Improve the installation guide for Android

### ISSUE THREAD

**Issue by guillaumevincent** [wix/react-native-navigation#3726]
It's probably not an issue but more a feedback about [Installing.md](https://wix.github.io/react-native-navigation/v2/#/docs/Installing)
I found really hard to follow the documentation.

For newcomers I found difficult to understand how to fix `MainActivity.java` and `MainApplication.java` and some issues with getJSMainModuleName implementation.
The flavor part is also disturbing coming at the beginning of the documentation referring to the end of the doc.

### Improvement

I created a simple [gist](https://gist.github.com/guillaumevincent/872a5798f48ff594465c4596aa7972cb) based on the installation of `react-native-navigation@alpha` and `react-native@0.56.0`

This guide starts with the idea that the user installed the application with the following step:

```
npm install -g react-native
react-native init myproject
```

### Environment
* React Native Navigation version: 2.0.2461
* React Native version: 0.56.0
* Platform(s) (iOS, Android, or both?): Android

Cheers

**Comment by guyca** [comment]
@guillaumevincent Thanks a lot for the feedback. The documentation and the installation guide in particular are a work in progress. Your feedback is spot on and the gist you posted looks great, would you mind PRing it? 🙏

**Comment by guillaumevincent** [comment]
@guyca I will create a PR and keep improve the flavor part

**Comment by guillaumevincent** [comment]
> There is still a little typo. Under the android/build.gradle , the ext.minSdkVersion comes 16 by default and has to be updated to 19.

Nice catch !

**Comment by guillaumevincent** [comment]
Here the pull request: https://github.com/wix/react-native-navigation/pull/3727

closes https://github.com/wix/react-native-navigation/pull/3695 and https://github.com/wix/react-native-navigation/pull/3712

Check typo and grammar, English is not my native language
